### PR TITLE
feat: Add to custom feed modal

### PR DESCRIPTION
**Pull request by AmarTrebinjac**
## Changes

Enables the "Add to custom feed" modal when selecting

TODO:

- [x] Add the option to follow a squad (not a part of the modal, but will add it here)

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-842 #done

### Preview domain
https://as-842-modal.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/profile/Header.tsx [hunk 1]
```diff
@@ -97,6 +100,27 @@ export function Header({
         />
         {!isSameUser && (
           <CustomFeedOptionsMenu
+            onAdd={(feedId) =>
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:103]: Wouldn't it be better to just do this inside the modal?
You don't need it anywhere else right?

File: packages/shared/src/components/profile/Header.tsx [hunk 2]
```diff
@@ -97,6 +100,27 @@ export function Header({
...
+              unfollow({
+                id: user.id,
+                entity: ContentPreferenceType.User,
+                entityName: user.username,
+                feedId,
+              })
+            }
+            onCreateNewFeed={() =>
+              router.push(
+                `/feeds/new?entityId=${user.id}&entityType=${ContentPreferenceType.User}`,
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:121]: Why do we pass the extra stuff?

File: packages/shared/src/components/profile/Header.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Header.tsx:103]: And then just pass entityId/entityType or something?
But

**Review by rebelchris — COMMENTED** [review]
Minor comments, nothing blocking but awaiting Ante's response as well.

File: packages/shared/src/components/profile/Header.tsx (hunk 1 above)
Comment (AmarTrebinjac) [reply packages/shared/src/components/profile/Header.tsx:103]: We could. Personally I like making components as dumb as possible, especially if they are being reused. Easier to change stuff slightly if we deem necessary. 

If you guys think its better to just pass the props to the modal and do all the logic inside instead, I can make that quick change

File: packages/shared/src/components/profile/Header.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/profile/Header.tsx:121]: so onSuccess after creating custom feed we can follow the entity automatically (happens in bookmarks as well)

File: packages/shared/src/components/profile/Header.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/profile/Header.tsx:103]: I am fine with leaving as is

**Review by capJavert — APPROVED** [review]
Nice work!

File: packages/shared/src/components/profile/Header.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Header.tsx:103]: Cool let's leave it :)

**Review by rebelchris — APPROVED** [review]

**Pull request merged**